firmware/Core/Src/uart_handler/uart_handler.c

```diff
@@ -77,6 +106,7 @@ void HAL_UART_RxCpltCallback(UART_HandleTypeDef *huart) {
 void UART_init_uart_handlers(void) {
     // enable the UART interrupt
     HAL_UART_Receive_IT(&hlpuart1, (uint8_t*) &UART_telecommand_buffer_last_rx_byte, 1);
+    HAL_UART_Receive_IT(UART_gps_port_handle, (uint8_t*) &UART_gps_buffer_last_rx_byte, 1);
```

Comment:
I think we actually want to enable and disable the GPS interrupt separately. Given that I've seen it spam constant 0's if it gets in that mode, we want to default to it being turned off, and only turn it on when we're ready to record data.

If it gets in the spamming-zeros mode, then the interrupt will be constantly triggered, and the satellite won't be able to do anything else. 

Let's remove this from here, and let's wrap the `HAL_UART_Receive_IT` call within the Interrupt Service Routine (ISR) in an `if (GPS_uart_interrupt_enabled) {}`. Then, let's have a function called `GPS_set_uart_interrupt_state`, which triggers `HAL_UART_Receive_IT`, and sets the `GPS_uart_interrupt_enabled` flag variable.

Reply:
Yeah, i think ive done an implementation of what you have requested. Let me know if it makes sense.